### ISSUE TITLE
Reduce the access key requirement to average transaction cost

### ISSUE THREAD

**Issue by ilblackdragon** [near/near-linkdrop#12]
Figure out if we can now use 0.01N or something similar for access key budget here https://github.com/near/near-linkdrop/blob/master/src/lib.rs#L18

Previously it was this high due to how much transaction would prepay if it would take next 100 blocks of receipts with constant price increase. Is this still the case?

**Comment by bowenwang1996** [comment]
> Is this still the case?

Yes. That part has not changed. Also it is 64 blocks instead of 100

**Comment by ilblackdragon** [comment]
Ok, so here is suggestion how to address this issue: 
 - we actually leave 0 allowance on the access key
 - but we remove it after 1st usage always (e.g. even if it's failed)
 - to make sure we can capture any call - we will need to wrap JSON parsing and other errors within functions (e.g. switch away from using `near_bindgen` for those functions).
 - if there are errors - we just return funds back to the original user (so need to record who sent funds)

**Comment by evgenykuzyakov** [comment]
> but we remove it after 1st usage always (e.g. even if it's failed)

It's possible to attach low gas amount, so the call will fail and we can't remove it within the call.

# Let the sender claim it back.

Alternative is to decrease the allowance from `1 NEAR` to minimum required balance in gas at near BASE gas price cost. This still supports a single valid function call, or multiple invalid one.

If the call fails for some reason (or the link is lost), the original sender will be able to claim it back, so you don't need the worry about funds from being locked. This is fairly simple to implement

**Comment by ilblackdragon** [comment]
What would happen if gas price goes up? User will not be able to execute such tx until gas prices lowers back?

Claiming back by sender make sense for sure, but I don't understand how this solves the problem at hand.
Btw, for frontend for sure would be good to know amount that will be held back to attach extra to what user sends and show to user how much they receive.

**Comment by ilblackdragon** [comment]
Fixed now
